base/sort.jl

```diff
@@ -426,7 +426,7 @@ function selectperm!{I<:Integer}(ix::AbstractVector{I}, v::AbstractVector,
                                  order::Ordering=Forward,
                                  initialized::Bool=false)
     if !initialized
-        @inbounds for i = 1:length(ix)
+        @inbounds for i in eachindex(ix)
```

Comment:
@timholy Maybe we should be a bit more careful before merging this kind of change? As we have seen, it can be more tricky than it sounds to check that all modifications make sense.